syntax = "proto3";

service RunExample {
    rpc Start(StartingExample) returns(StartingStatus){}
    rpc QueryStatus(RunningExample) returns(RunningStatus){}
    rpc GetResult(CompletedExample) returns(stream ResultFile){}
}

message StartingExample {
    int64 example_id = 1;
    string example_kind = 2;
    string created_at = 3;
}

message StartingStatus {
    int64 example_id = 1;
    string example_name = 2;
    bool is_successful = 3;
}

message RunningExample{
    int64 example_id = 1;
    string example_name = 2;
}

message RunningStatus{
    int64 example_id = 1;
    string example_name = 2;
    bool is_completed = 3;
    string log_file_name = 4;
    string log_file_content = 5;
}

message CompletedExample{
    int64 example_id = 1;
    string example_name = 2;
    string example_kind = 3;
}

message ResultFile{
    string file_name = 1;
    bytes file_content = 2;
}